http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -209,6 +177,41 @@ class LeoPubsubMessageSubscriber[F[_]](
       _ <- F.delay(
         event.consumer.ack()
       )
+      _ <- recordMessageMetric(event)
+    } yield ()
+
+  private def processFailed(ctx: AppContext, event: Event[LeoPubsubMessage], e: Throwable)(implicit ev: Ask[F, AppContext]): F[Unit] = {
```

Comment:
@Qi77Qi - I wanted to bounce this refactor-ing off of you. I think it follows the cats-effect syntax. Trying to make the processing more testable ... working on this now.

Reply:
this is great...for testing, u can do `private[<whatever package>]`, that way you'll be able to access this in unit tests...